### ISSUE TITLE
Plugin: Configure TechDocs plugin

### ISSUE THREAD

**Issue by tumido** [operate-first/service-catalog#13]
Tech Docs should be installed by default, verify that and enable it on an example component.

https://backstage.io/docs/features/techdocs/techdocs-overview

**Comment by SamoKopecky** [comment]
We need to decide on two things for the `techdocs` configuration.
1. `Techdocs` are generated from `.md` files in a repository that contains `catalog-info.yaml` and `mkdocs.yaml` files. More info [here](https://backstage.io/docs/features/techdocs/creating-and-publishing#enable-documentation-for-an-already-existing-entity). Example of a repository [here](https://github.com/skopecky-test/example-service/).
   - We either have to link one repository per service (like [this example](https://backstage.io/docs/features/software-catalog/configuration#static-location-configuration)).
   - Or we can set up [GitHub integration](https://backstage.io/docs/integrations/github/locations#configuration), which only requires either a personal access token or a GitHub App. This will discover all of the repositories that contain `catalog-info.yaml` file and add it to the service-catalog ([docs](https://backstage.io/docs/integrations/github/discovery#configuration)). This means if someone wants to add a new service to our service-catalog all he needs to do is add a `catalog-info.yaml` file to his repository and it will be automatically picked up and added. This would of course only work for repositories/organizations that the token/app has access to.
2. `Techdocs` building process 
   - The docs need to be built somewhere, we can install local dependencies (list bellow) and build them inside the backstage container.
      - [mkdocs](https://pypi.org/project/mkdocs/)
      - [mkdocs-techdocs-core](https://pypi.org/project/mkdocs-techdocs-core/1.0.1/)
    - Or we can utilize a docker image for building, this container will be run inside our backstage container.
    
There is also an option to publish the generated docs files to external storage ([docs](https://backstage.io/docs/features/techdocs/using-cloud-storage)). I think it's fine if we don't publish the built docs anywhere and just store them locally for now.

**Comment by tumido** [comment]
Ad 1)
Link to techdocs within the service manifest should be preferred. It supports independence of each service and self governance.

Ad 2) Let's go the simplest path available for now and build it via 1) or 2) option, whatever is easier to achieve. CI/CD and external storage proposed in 3) can turn into a complex beast - possible backlog item.

**Comment by SamoKopecky** [comment]
I've done some more research on the possible solutions that are using cloud storage for publishing:

## 1. Each service will host their own docs on a separate repository and they will build&publish docs to S3 storage (recommended backstage setup)
- Using GitHub Actions, where each repository will host their own CI/CD
- Each repo will need the access to S3 bucket credentials
- Setup required for each newly added service
  
Architecture diagram:
![image](https://user-images.githubusercontent.com/44006847/179956957-f56d5eeb-aa3e-4c0e-a986-259d09ef846c.png)

## 2. Each service will host their own docs in their repository (or multiple services in one repository) and we will build&publish the docs to storage
 - GitHub actions will listen for [`repository_dispatch`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#repository_dispatch) events of each repository hosting docs
 - Multiple directories can exists in one repo, each directory has the name of the component so we know for which component to publish docs
 - Only one repository needs to have s3 bucket credentials
 - The repositories hosting source docs files will only setup action for pushing [`repository_dispatch`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#repository_dispatch) event when docs are updated

Architecture diagram:
![image](https://user-images.githubusercontent.com/44006847/179958240-bf9d010b-689c-4d8c-bed5-956935a45d60.png)

## 3. Each service will host their own docs in their repository (or multiple services in one repository) and we will build&publish the docs to storage using a probot application
- Basically the same as number 2 but using a probot app listening to webhooks instead of github actions listening to `repository_dispatch` events
- Separate instance from backstage running and listening for push webhook events on repositories that host source docs files 
- Only a webhook setup is required in service repositories



Architecture diagram:
![image](https://user-images.githubusercontent.com/44006847/179958679-a1b71d98-7d2a-4f12-a35e-4072996c1f93.png)

**Comment by SamoKopecky** [comment]
Techdocs are function and can be verified here: https://service-catalog.operate-first.cloud/catalog/default/component/grafana-public/docs
